ion-schema/src/test/kotlin/com/amazon/ionschema/reader/internal/constraints/LogicConstraintsReaderTest.kt

```diff
@@ -0,0 +1,79 @@
+package com.amazon.ionschema.reader.internal.constraints
+
+import com.amazon.ion.IonStruct
+import com.amazon.ion.system.IonSystemBuilder
+import com.amazon.ionschema.model.Constraint
+import com.amazon.ionschema.model.ExperimentalIonSchemaModel
+import com.amazon.ionschema.model.TypeArgument
+import com.amazon.ionschema.reader.internal.ReaderContext
+import com.amazon.ionschema.reader.internal.TypeReader
+import io.mockk.every
+import io.mockk.mockk
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.ValueSource
+
+@ExperimentalIonSchemaModel
+class LogicConstraintsReaderTest {
```

Comment:
🗺️ The tests here are intentionally sparse. `canRead` is completely covered, but tests for `read` cover only a few cases since comprehensive testing is handled with the `ion-schema-tests` test cases.